VS_Meadow_Extension/VS_Meadow_Extension.Shared/MeadowDeployProvider.cs

```diff
@@ -143,25 +150,31 @@ private async Task<string> GetOutputPathAsync(string filename)
             return outputPath;
         }
 
-        private static void MeadowConnection_DeviceMessageReceived(object sender, (string message, string source) e)
+        private static async void MeadowConnection_DeviceMessageReceived(object sender, (string message, string source) e)
         {
-            _ = outputLogger.ReportDeviceMessage(e.message);
+            await ThreadHelper.JoinableTaskFactory.RunAsync(async () =>
```

Comment:
Check this - I think the JoinableTaskFactory is handled within ReportDeviceMessage

Reply:
In this case it wasn't, but I've moved the call inside ReportDeviceMessage, to make it consistent with all the other handlers.